documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
+
+- [ ] **M1003**: List operations *MUST* be prefixed with "List" and *MUST NOT* be named as "GetXXXX" operations.
+
+- [ ] **M1004**: List operations *MUST* use consistent names and semantics. List operations *MUST* NOT use any other names. Consistent names and semantics are:
+
+  - `List()` - lists all resources under a subscription.
+  - `ListByResourceGroup()` - list all resources in a resource group within a subscription.
+  - `ListByParent()` - where "Parent" is a context specific suffix. It lists all resource under a parent.
+
+- [ ] **M1005**: Get operations *MUST* use the METHOD name "Get"
+
+- [ ] **M1006**: PUT operations *MUST* use the METHOD Name "Create"
+
+- [ ] **M1007**: PATCH operations *MUST* use the METHOD Name "Update"
+
+- [ ] **M1008**: POST operations *MUST* use a METHOD Name that matches the semantic operation on the resource, for example "Start" or "Stop"
 
-&#x25a2; *M1001*: List operations MUST be prefixed with "List" and MUST not be named as "GetXXXX" operations.
+- [ ] **M1009**: DELETE operations *MUST* use the METHOD Name "Delete"
 
-&#x25a2; *M1002*: List operations MUST use consistent names and semantics. List operations MUST NOT use any other names. Consistent names and semantics are:
+- [ ] **M1010**: Terminology in Swagger and Azure Portal *MUST* be the same.
 
-- `List()` - lists all resources under a subscription.
-- `ListByResourceGroup()` - list all resources in a resource group within a subscription.
-- `ListByParent()` - where "Parent" is a context specific suffix. It lists all resource under a parent.
+- [ ] **M1011**: Resource Model name *MUST* be same as singular form of the resource type. Like for virtualMachines resource type it should be VirtualMachine.
 
-&#x25a2; *M1003*: Terminology in Swagger and Azure Portal MUST be the same.
+- [ ] **M1012**: Parameter names in the apis *MUST* be name of the resource like virtualMachines\{virtualMachine} or virtualMachines\{virtualMachineName} both are good example but virtualMachines\{vm} is bad example. 
 
+- [ ] **M1013**: Any action name *MUST* be same as what is defined in the Rest APIs, like regenerateAdminKeys should be regenerateAdminKeys in the Swagger as well, where as regenerateKeys is bad example.
 
 
 ## Operation Semantics - Swagger Checklist ##
 
-&#x25a2; *M2000*: A Swagger specification MUST be accompanied with a document that describes management scenarios that can be realized using operations described in the Swagger.
+### MUST
 
-&#x25a2; *M2001*: Read only parameters MUST be labeled as read only in Swagger.
+- [ ] **M2000**: TODO: A Swagger specification *MUST* be accompanied with a document that describes management scenarios (what does this mean? x-ms-examples or a multi operation scenario?) that can be realized using operations described in the Swagger.
```

Comment:
In what format is that supposed to be delivered? Who are the intended consumers (doc team, SDK teams etc.)?